pkg/v1/tkg/tkgctl/featuregate_helper.go

```diff
@@ -40,14 +42,21 @@ func (fg *featureGateHelper) FeatureActivatedInNamespace(reqContext context.Cont
 	scheme := clusterClient.GetClientSet().Scheme()
 	_ = v1alpha1.AddToScheme(scheme)
 
-	// If requested feature is `ClusterClassFeature` or `TKCAPIFeature` it gets decided on different criteria at the moment
-	// This is because we are unable to query Featuregates for the devops users on the TKGS clusters because of the permission restrictions
-	// TODO: Fix this logic to rely on actual featuregates once permission issue for the devops user has been resolved
-	switch feature {
-	case constants.ClusterClassFeature:
-		return fg.isClusterClassFeatureActivated(clusterClient, feature, namespace)
-	case constants.TKCAPIFeature:
-		return fg.isTKCFeatureActivated(clusterClient, feature, namespace)
+	isPacific, err := clusterClient.IsPacificRegionalCluster()
+	if err != nil {
+		return false, errors.Wrap(err, "error determining Tanzu Kubernetes Cluster service for vSphere management cluster ")
+	}
+
+	if isPacific {
+		// If requested feature is `ClusterClassFeature` or `TKCAPIFeature` it gets decided on different criteria at the moment
+		// This is because we are unable to query Featuregates for the devops users on the TKGS clusters because of the permission restrictions
+		// TODO: Fix this logic to rely on actual featuregates once permission issue for the devops user has been resolved
```

Comment:
Q: Is there a GH issue for this TODO?

Reply:
I have filled: https://github.com/vmware-tanzu/tanzu-framework/issues/3052

Reply:
Unfortunately, it's kind of hard to write unit tests at this level of cobra commands but we have unit tests for internal functions being invoked here.